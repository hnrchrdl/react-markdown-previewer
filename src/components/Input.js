import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { createUpdateInputAction } from '../actions/input';

const InputContainer = styled.div`
  position: relative;
  width: 100%;
  box-sizing: border-box;
`;
const Textarea = styled.textarea`
  width: 100%;
  height: 300px;
  box-sizing: border-box;
  border: 3px solid #eee;
  padding: 20px;
  outline: none;
`;

class Input extends React.Component {
  handleInputChange = (event: SyntheticEvent) => {
    this.props.updateInputValue(event.target.value);
  };
  render() {
    return (
      <InputContainer>
        <Textarea
          id="editor"
          placeholder="Begin typing here ..."
          value={this.props.inputValue}
          onChange={this.handleInputChange}
        />
      </InputContainer>
    );
  }
}

const mapStateToProps = state => ({ inputValue: state.io.input });
const mapDispatchToProps = dispatch => ({
  updateInputValue(value) {
    dispatch(createUpdateInputAction(value));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Input);
