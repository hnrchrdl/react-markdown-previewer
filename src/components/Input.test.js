import React from 'react';
import Input from './Input';
import renderer from 'react-test-renderer';
import store from '../store';

test('App renders and matches snapshot', () => {
  const component = renderer.create(<Input store={store} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
