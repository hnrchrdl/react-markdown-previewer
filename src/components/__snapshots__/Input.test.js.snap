// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders and matches snapshot 1`] = `
<div
  className="sc-bdVaJa jLZiFJ"
>
  <textarea
    className="sc-bwzfXH hxIJCZ"
    id="editor"
    onChange={[Function]}
    placeholder="Begin typing here ..."
    value="
    # a header (H1 size)
    ## a sub header (H2 size)
    [a link]
    \`inline code\`  
    \`\`\`a code block\`\`\`  
    1. a list item
    > a blockquote  
    ![an image](https://cdn.vox-cdn.com/thumbor/YqrPXWiAFv9TuM0tUi_i1pZEPyE=/0x0:512x512/920x0/filters:focal(0x0:512x512):format(webp)/cdn.vox-cdn.com/uploads/chorus_asset/file/6808705/Cat_101.0.png)
    **and bolded text**"
  />
</div>
`;
